dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,167 @@
+
+    def download_file_to_storage(self, file_name):
+        # function to download the file and add it to kernel file system
+        file_output_path = join(self.local_dir, file_name)
+        if exists(file_output_path):
+            print('- [{}] file already exists. Skipping download.'.format(file_name))
+            return
+
+        print('- Fetching [{}] ...'.format(file_name))
+        x = self.files_list[file_name]
+
+        if file_name == 'BlueConfig':
+            file_content = x.raw().read()
+            bc_str = file_content.decode('utf-8')
+            writable_content = bc_str.replace(self.ORIGIN_CIRCUIT_PATH, self.circuit_dir)
+            writable_content = writable_content.replace(self.ORIGIN_DIR_PATH, self.local_dir)
+
+            with open(file_output_path, 'w') as fd:
+                fd.write(writable_content)
+
+        else:
+            x.download(str(file_name)) # moves to home always
+            move(file_name, file_output_path)
+
+        print('- [{}] downloaded'.format(file_name))
+
+    def download_blueconfig(self, file_name='BlueConfig'):
+        self.download_file_to_storage(file_name)
+
+    def download_report(self):
+        reports = [x for x in self.files_list if '.bbp' in x]
+        if not reports:
+            print('No reports were found')
+            return
+        for report in reports:
+            self.download_file_to_storage(report)
+
+    def download_out_dat(self):
+        self.download_file_to_storage('out.dat')
+
+
+class FetchMultipleResults:
+
+    values = []
+
+    def __init__(self, token, sim_list_ids, base_working_directory=None):
+
+        total_sims = len(sim_list_ids)
+        for idx, sim in enumerate(sim_list_ids):
+            print('({}/{})'.format(idx + 1, total_sims))
+            self.values.append(Results(token, sim, base_working_directory))
+
+
+class JobFinder:
+
+    jobs = []
+    sorted_jobs = None
+
+    def __init__(self, token):
+        tr = unicore_client.Transport(token)
+        client = unicore_client.Client(tr, UNICORE_ENDPOINT)
+        self.jobs = client.get_jobs()
+
+    def show_list(self, amount_jobs_to_show=None):
+        full_jobs_list = []
+        if amount_jobs_to_show is None:
+            amount_jobs_to_show = 1000
+
+        for index, job in enumerate(self.jobs[:amount_jobs_to_show]):
+            raw_time = job.properties['submissionTime']
+            full_time = datetime.strptime(raw_time, '%Y-%m-%dT%H:%M:%S%z')
+            date = full_time.strftime('%d-%m-%Y %H:%M')
+            name = job.properties['name']
+            job_id = basename(job.properties['_links']['self']['href'])
+            full_jobs_list.append([index, name, date, job_id])
+            print('Fetching jobs ({}/{})'.format(index + 1, amount_jobs_to_show or len(jobs) + 1), end = '\r')
+
+        columns = ['index', 'name', 'date', 'job_id']
+        self.sorted_results = pd.DataFrame(full_jobs_list, columns=columns).sort_values(by=['date'])
```

Comment:
```suggestion
        columns = ['name', 'date', 'job_id']
        self.sorted_results = pd.DataFrame(full_jobs_list, columns=columns).reset_index().sort_values(by=['date'])
```
But check that it works, I'm not so sure :D